File: src/main/java/no/nav/veilarbvedtaksstotte/controller/dto/KodeverkDTO.java
package no.nav.veilarbvedtaksstotte.controller.dto;

import lombok.Data;
import lombok.experimental.Accessors;
import no.nav.veilarbvedtaksstotte.domain.vedtak.HovedmalDetaljert;
import no.nav.veilarbvedtaksstotte.domain.vedtak.InnsatsgruppeDetaljert;

@Data
@Accessors(chain = true)
public class KodeverkDTO {
    InnsatsgruppeDetaljert[] innsatsgrupper;
    HovedmalDetaljert[] hovedmal;

    public KodeverkDTO() {
        this.innsatsgrupper = InnsatsgruppeDetaljert.values();
        this.hovedmal = HovedmalDetaljert.values();
    }
}

File: src/main/java/no/nav/veilarbvedtaksstotte/controller/dto/KodeverkDTO.kt
package no.nav.veilarbvedtaksstotte.controller.dto

import no.nav.veilarbvedtaksstotte.domain.vedtak.HovedmalMedOkeDeltakelse
import no.nav.veilarbvedtaksstotte.domain.vedtak.Innsatsgruppe
import no.nav.veilarbvedtaksstotte.domain.vedtak.InnsatsgruppeDetaljert.ArenaInnsatsgruppeKode

data class KodeverkDTO(
    val innsatsgrupper: List<InnsatsgruppeKodeverkDTO>,
    val hovedmal: List<HovedmalKodeverkDTO>
)

data class InnsatsgruppeKodeverkDTO(
    val kode: Innsatsgruppe,
    val beskrivelse: String,
    val arenakode: ArenaInnsatsgruppeKode
)

data class HovedmalKodeverkDTO(
    val kode: HovedmalMedOkeDeltakelse,
    val beskrivelse: String
)
